national_rail/transform.py

```diff
@@ -0,0 +1,153 @@
+
+    current_time = datetime.now()
+    time_diff = current_time - creation_time
+
+    return time_diff <= timedelta(minutes=5)
+
+
+def process_pt_incidents(incidents: list[ET.Element], namespaces: dict) -> list[dict]:
+    """ Extracts relevant data for each incident reported. """
+    dataset = []
+
+    for incident in incidents:
+
+        creation_time = convert_to_datetime(find_text_element(
+            incident, 'ns:CreationTime', namespaces))
+
+        if not check_creation_within_last_5_minutes(creation_time):
+            # remove not to allow all incidents
+            logging.info("No new incidents found within the last 5 minutes")
+            break
+
+        incident_number = find_text_element(
+            incident, 'ns:IncidentNumber', namespaces)
+
+        operator_codes = find_all_text_elements(
+            incident, "ns:Affects/ns:Operators/ns:AffectedOperator/ns:OperatorRef", namespaces)
+
+        start_time = find_text_element(
+            incident, 'ns:ValidityPeriod/com:StartTime', namespaces)
+        end_time = find_text_element(
+            incident, 'ns:ValidityPeriod/com:EndTime', namespaces)
+
+        is_planned = find_text_element(incident, 'ns:Planned', namespaces)
+
+        summary = find_text_element(incident, 'ns:Summary', namespaces)
+
+        description = convert_html_to_text(find_text_element(
+            incident, 'ns:Description', namespaces))
+
+        uri = find_text_element(
+            incident, "ns:InfoLinks/ns:InfoLink/ns:Uri", namespaces).replace('/n', " ").strip()
+
+        routes_affected = convert_html_to_text(find_text_element(
+            incident, "ns:Affects/ns:RoutesAffected", namespaces))
+
+        data = {
+            'incident_number': incident_number,
+            'operator_codes': operator_codes,  # list
+            'creation_time': creation_time,
+            'start_time': start_time,
+            'end_time': end_time,
+            'is_planned': is_planned,
+            'summary': summary,
+            'description': description,
+            'uri': uri,
+            'routes_affected': routes_affected
+        }
+        dataset.append(data)
+
+    return dataset
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO,
+                        format="%(asctime)s - %(levelname)s - %(message)s")
+
+    logging.info("Transformation of NationalRail has began")
+
+    nr_namespaces = {'ns': 'http://nationalrail.co.uk/xml/incident',
+                     'com': 'http://nationalrail.co.uk/xml/common'}
+
+    national_rail_data = read_data_from_file("data.xml")
+
+    national_rail_tree_root = reverse_tree(load_tree_root(national_rail_data))
+
+    all_incidents = get_incidents(national_rail_tree_root, nr_namespaces)
+
+    incidents_dataset = process_pt_incidents(all_incidents, nr_namespaces)
+
+    logging.info("Transformation of NationalRail completed successfully")
```

Comment:
You shouldn't log here as it will not be used in the pipeline file, move it to a function